crates/wit-parser/src/ast/resolve.rs

```diff
@@ -364,6 +399,20 @@ impl<'a> Resolver<'a> {
                         assert!(prev.is_none());
                         let prev = names.insert(i.name.name, item);
                         assert!(prev.is_none());
+                        for nest_item in i.items.iter() {
+                            if let InterfaceItem::Nest(n) = nest_item {
+                                if package_items.insert(n.name.name, n.name.span).is_some() {
+                                    bail!(Error::new(
+                                        n.name.span,
+                                        format!("duplicate item named `{}`", n.name.name),
+                                    ))
+                                }
+                                let prev = decl_list_ns.insert(n.name.name, ());
+                                assert!(prev.is_none());
+                                let prev = order.insert(n.name.name, Vec::new());
+                                assert!(prev.is_none());
+                            }
+                        }
```

Comment:
I'm not sure if this is quite right because this seems like it would be an `interface`-level validation rather than a package-level validation which I think this loop is doing.